staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go

```diff
@@ -268,3 +282,43 @@ func (f *FieldManager) stripFields(managed fieldpath.ManagedFields, manager stri
 
 	return managed
 }
+
+// NameDefaulter makes sure the request option and meta field match if both are set. If neither are set it defaults to the userAgent or errors
+func NameDefaulter(option, field, userAgent string) (string, error) {
+	if len(option) != 0 {
+		if len(field) != 0 && option != field {
+			return "", fmt.Errorf("fieldManager option must match metadata fieldManager or be unset")
```

Comment:
`the fieldManager of the provided object does not match the fieldManager sent on the request` (copied from namespace error)?

Reply:
no, didn't have time to look it up yet.